ee/api/performance_events.py

```diff
@@ -0,0 +1,126 @@
+    response_end = serializers.FloatField()
+    decoded_body_size = serializers.IntegerField()
+    encoded_body_size = serializers.IntegerField()
+    initiator_type = serializers.CharField()
+    next_hop_protocol = serializers.CharField()
+    render_blocking_status = serializers.CharField()
+    response_status = serializers.IntegerField()
+    transfer_size = serializers.IntegerField()
+    largest_contentful_paint_element = serializers.CharField()
+    largest_contentful_paint_render_time = serializers.FloatField()
+    largest_contentful_paint_load_time = serializers.FloatField()
+    largest_contentful_paint_size = serializers.FloatField()
+    largest_contentful_paint_id = serializers.CharField()
+    largest_contentful_paint_url = serializers.CharField()
+    dom_complete = serializers.FloatField()
+    dom_content_loaded_event = serializers.FloatField()
+    dom_interactive = serializers.FloatField()
+    load_event_end = serializers.FloatField()
+    load_event_start = serializers.FloatField()
+    redirect_count = serializers.IntegerField()
+    navigation_type = serializers.CharField()
+    unload_event_end = serializers.FloatField()
+    unload_event_start = serializers.FloatField()
+
+
+class PerformanceEvents:
+    @classmethod
+    def query(cls, team_id: int, session_id: str, pageview_id: Optional[str] = None) -> List[Dict]:
+        query = """
+                select * from performance_events
+                prewhere team_id = %(team_id)s
+                and session_id = %(session_id)s
+                order by timestamp
+                """
+
+        if pageview_id:
+            query += " and pageview_id = %(pageview_id)s"
+        ch_results = sync_execute(
+            query,
+            {"team_id": team_id, "session_id": session_id, "pageview_id": pageview_id},
+        )
+
+        columns = [
+            col.strip() for col in _column_names_from_column_definitions(PERFORMANCE_EVENT_COLUMNS).split(", ") if col
+        ]
+        columnized_results = []
+        for result in ch_results:
+            columnized_item = {}
+            for index, column in enumerate(result):
+                if index < len(columns):
+                    columnized_item[columns[index]] = column
+            columnized_results.append(columnized_item)
+
+        return columnized_results
+
+
+class PerformanceEventsViewSet(StructuredViewSetMixin, viewsets.GenericViewSet):
+    serializer_class = PerformanceEventSerializer
+    permission_classes = [
+        IsAuthenticated,
+        PremiumFeaturePermission,
+        ProjectMembershipNecessaryPermissions,
+        TeamMemberAccessPermission,
+    ]
+    premium_feature = AvailableFeature.RECORDINGS_PERFORMANCE
+
+    def get_queryset(self):
+        return None
+
+    def list(self, request: request.Request, *args, **kwargs) -> Response:
+        session_id = request.GET.get("session_id")
+        pageview_id = request.GET.get("pageview_id")
```

Comment:
Shall we include the timestamp bounds in the API call now?

eg. `before any timestamp in the session + 48 hours` and `after any timestamp in the session -48 hours`

Reply:
Yeah we probably should... I think the user should provide it though, otherwise the lookup is double-y long. But yeah I'll add `date_from` and `date_to` like the other APIs